### PR TITLE
Bugfix 591:  Scheduled exports run with each service run until missed executions are recovered

### PR DESCRIPTION
**Pull request by rivaldi8**
@codinguser please, review this as it's hackish and I might have got something wrong. I have another bug depending on this one so I'll be working on it meanwhile.

Fixes https://github.com/codinguser/gnucash-android/issues/591

### CODE REVIEW

File: app/src/main/java/org/gnucash/android/service/ScheduledActionService.java [hunk 1]
```diff
@@ -175,14 +175,32 @@ private static int executeBackup(ScheduledAction scheduledAction, SQLiteDatabase
...
             //wait for async task to finish before we proceed (we are holding a wake lock)
             new ExportAsyncTask(GnuCashApplication.getAppContext(), db).execute(params).get();
-            scheduledAction.setExecutionCount(++executionCount);
+            ++executionCount;
         } catch (InterruptedException | ExecutionException e) {
             Crashlytics.logException(e);
             Log.e(LOG_TAG, e.getMessage());
         }
+
+        // HACK: Add missed runs to executionCount so computeNextScheduledExecutionTime()
```
Comment (codinguser) [inline app/src/main/java/org/gnucash/android/service/ScheduledActionService.java]: If this fixes the problem, then we can live with it for now. As long as the meaning of the executionCount is not changed.
In the future backups & exports may be separated from scheduled transactions and then we will have to revisit this.

File: app/src/main/java/org/gnucash/android/model/ScheduledAction.java [hunk 2]
```diff
@@ -322,6 +327,11 @@ public int getExecutionCount(){
 
     /**
      * Sets the number of times this scheduled action has been executed
+     *
+     * <p><b>Important:</b> when updating this field include not only the real
```
Comment (rivaldi8) [inline app/src/main/java/org/gnucash/android/model/ScheduledAction.java]: Yes, I agree. It was a last minute change and I really don't like it either. Specially taking into account that it only applies to scheduled exports.

File: app/src/test/java/org/gnucash/android/test/unit/service/ScheduledActionServiceTest.java [hunk 3]
```diff
@@ -280,10 +280,25 @@ public void recurringTransactions_shouldHaveScheduledActionUID(){
...
+     * useful, so just one should be done.</p>
+     *
+     * <p><i>Note</i>: the execution count will include the missed runs
+     * as computeNextScheduledExecutionTime depends on it.</p>
+     */
     @Test
     public void scheduledBackups_shouldRunOnlyOnce(){
         ScheduledAction scheduledBackup = new ScheduledAction(ScheduledAction.ActionType.BACKUP);
-        scheduledBackup.setStartTime(new DateTime(2016, 2, 17, 17, 0).getMillis());
+        scheduledBackup.setStartTime(LocalDateTime.now()
```
Comment (rivaldi8) [inline app/src/test/java/org/gnucash/android/test/unit/service/ScheduledActionServiceTest.java]: Yes, it really nice to test for certain time-dependant conditions :)

File: app/src/main/java/org/gnucash/android/service/ScheduledActionService.java (hunk 1 above)
Comment (rivaldi8) [inline app/src/main/java/org/gnucash/android/service/ScheduledActionService.java]: Yes, my initial intention was to separate scheduled exports and transactions into different classes but again, it depended on having an XML pull parser for importing. That's why I did this temporal hack. So I guess I'll start with the parser as soon as I fix these bugs.

Comment (codinguser) [inline app/src/main/java/org/gnucash/android/service/ScheduledActionService.java]: We'll separate them with bigger db schema changes down the line.

**Pull request closed without merging**